samples/src/Cache/CacheAspect.cs

```diff
@@ -14,29 +14,33 @@ public abstract class CacheAttribute : Attribute
         public abstract ObjectCache Cache { get; }
 
         public abstract CacheItemPolicy Policy { get; }
+
+        public bool PerInstanceCache { get; set; }
     }
 
     [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
     public class MemoryCacheAttribute : CacheAttribute
     {
         private static readonly MemoryCache _cache = new MemoryCache("aspect_builtin_memory_cache");
+
         private readonly uint _seconds;
 
-        public MemoryCacheAttribute(uint seconds)
+        public MemoryCacheAttribute(uint seconds, bool perInstanceCache = false)
         {
             _seconds = seconds;
+            PerInstanceCache = perInstanceCache;
         }
 
         public override ObjectCache Cache => _cache;
+
         public override CacheItemPolicy Policy => new CacheItemPolicy() { AbsoluteExpiration = DateTimeOffset.UtcNow.AddSeconds(_seconds) };
     }
 
 
     [Aspect(Scope.Global)]
     public class CacheAspect
     {
-        private static readonly Type _voidTaskResult = Task.FromException(new Exception()).GetType();
-        private static readonly object _nullmarker = new object();
+        private static readonly object NullMarker = new object();

```

Comment:
Why naming change?

Reply:
CamelCase is a standard recommended for static fields: https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/d53b55ey(v=vs.71)?redirectedfrom=MSDN

Reply:
In this project we use CamelCase only for public fields, for private we use underscore prefix. I know, it is my fault I shoud've written the contribution guide long ago. That is a good push for me.